### ISSUE TITLE
Radio Button Group does not have an associated label prop which is required by screen readers

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#66475]
### User Experience

Screen reader users will get impacted if the label of any control or radio buttons are not announced by the screen reader. Users will find it difficult to understand the control purpose if label is not announced and not able to take actions on controls.

### Repro steps

Go to the [RadioButtonGroup](https://developers.grafana.com/ui/latest/index.html?path=/story/forms-radiobuttongroup--radio-buttons) in the Storybook UI. Turn on your screen reader. Observe that no label is read out associated with the radio options when you first enter them.

e.g. In this case the screen reader would need to read out "Prometheus, selected, radio button, 1 of 3, **_Full width, group_**" _(currently it doesn't read out Full width)_
![image](https://user-images.githubusercontent.com/6906380/233142919-2239d8fa-9505-409e-b6a7-962700711884.png)

A practical example in the application:
https://user-images.githubusercontent.com/6906380/231788398-d4016577-8619-45bb-ba64-933ee0fb27ac.mov

### Severity
`SEV 2`

### MS Sheet ID
[256757](https://docs.google.com/spreadsheets/d/1kB-Sz3lvfje93HLVIx4UhDvwgU-J7k1PVN70ryQkc4M/edit#gid=888389206&range=A4)

**Comment by khushijain21** [comment]
I'd like to work on this, is it still open?

**Comment by ckbedwell** [comment]
Hi @khushijain21! Yeah, this is free to be picked up. 😃  If you have any further questions or want clarification feel free to reach out and I'll do what I can to help.

**Comment by ckbedwell** [comment]
Ah, so maybe it wasn't made clear enough in the issue. I've updated the description to make it clearer. The labels for the individual choices are read out as you noted (they aren't for the icons version, mind, but that's another bug...) but the over-arching label associated with the group isn't read out.

You can see an example of it working correctly on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) and also on [Deque University](https://dequeuniversity.com/library/aria/radio-and-radio-group).
